comRamona/assignment4/NumberOfIslands.java

```diff
@@ -0,0 +1,49 @@
+package assignment4;
+
+/**
+ * You are given a 2-dimensional map of tiles. Each tile is either land or water. You have to write a function that
+ * counts the number of islands. Two tiles belong to the same island if they are both land and are adjacent horizontally
+ * or vertically, but not diagonally.The input to your function is the number of rows, number of columns, and a
+ * 2-dimensional array of booleans, where false means water and true means land. The function should simply return the
+ * number of islands.
+ */
+public class NumberOfIslands {
+
+    /**
+     * Computes the number of islands in a grid of tiles
+     *
+     * @param rows    number of rows
+     * @param columns number of columns
+     * @param grid    given grid of tiles
+     * @return number of islands
+     */
+    public static int countIsland(int rows, int columns, boolean[][] grid) {
+        int count = 0;
```

Comment:
You can infer rows and columns from the grid itself, so do not need to parse them in.